dev/sparktestsupport/modules.py

```diff
@@ -542,6 +542,10 @@ def __hash__(self):
         "pyspark.testing.utils",
         "pyspark.testing.pandasutils",
     ],
+    excluded_python_implementations=[
+        "PyPy"  # Skip these tests under PyPy since they require numpy, pandas, and pyarrow and
```

Comment:
Well, I'd prefer to handle these missing packages in the test cases.

Reply:
cc @itholic

Reply:
The reason why this place is different from other places is that there are some `doctests` in `pyspark.testing.utils`. When I tried adding `@unittest.skipIf(not have_pyarrow, pyarrow_requirement_message)` to the main method, it didn't seem to work.

Reply:
What about using `# doctest: +SKIP`? The functions in `pyspark.testing.utils` are also have their corresponding UTs, so I think it should be fine to skip the doctest.

Reply:
Okay, Let me give it a try.

Reply:
I have added `# doctest: +SKIP` to all the doctests using method `assertDataFrameEqual` in `pyspark.testing.utils`, and based on the current GA results, they have all passed.

Reply:
Cool. Looks good to me.

Reply:
can we remove `PyPy` here, and make the tests skipped properly?

Reply:
Yeah, after adding `# doctest: +SKIP`, GA has passed.

Reply:
Wait a moment, I forgot to remove it. I'll test it.

Reply:
By the way, I remember that after I added `# doctest: +SKIP`, I tested it `locally` once.
Let it run GA again. 😄

Reply:
@HyukjinKwon `Pyspark tests` have passed.